### PR TITLE
fix: trainEventRec test dataloader mocks

### PR DESCRIPTION
**Pull request by wonb1te**
## Summary

Fixes the trainEventRec tests so the mocked dataloaders return a fresh iterator each time they are iterated. This removes the false PyTorch warning about lr_scheduler.step() being called before optimizer.step() without changing the production trainer logic.

---

## Why is this change necessary?

trainEventRec.main() calls iter(train_loader) more than once: first to inspect a batch and infer num_tags, and then again during the training loop. The old test mocks reused a single iterator via __iter__.return_value, so the first read exhausted it before training began. That caused the training loop to skip, which made the scheduler warning appear even though the real code path was correct.

Closes #

---

## Changes

Updated packages/data_ml/tests/test_trainEventRec.py
  - Changed mocked dataloader setup in test_main_trains_from_scratch
  - Changed mocked dataloader setup in test_main_loads_checkpoint
  - Replaced __iter__.return_value = iter(...) with __iter__.side_effect = lambda: iter(...) so each iteration gets a fresh iterator
  
---

## Testing

1. cd packages/data_ml
2. uv run pytest tests/test_trainEventRec.py -q

### CODE REVIEW

**Review by Ekinn7188 — APPROVED** [review]
Looks great! Now we can be sure that our PyTorch models train properly. Order of backpropagation and scheduler steps are vital to ensure everything trains as expected.

Hacking Beavis

**Review by Manj0t — APPROVED** [review]
LGTM. This helps out a lot to make sure there are no issues with the training.
